include/optimization/cutting_plane.h

```diff
@@ -0,0 +1,1049 @@
+            min1 = min2;
+            min2 = t;
+        }
+
+        std::pair<NT, NT> bpair;
+
+        // this point will be the end point of the segment of the minimizing point, that lies in the polytope after the cut
+        // we will use it to get an interior point to start the random walk at the next phase
+        Point boundaryMin1 = min1;
+        Point boundaryMin2 = min2;
+
+        // begin sampling
+
+        for (unsigned int i = 1; i <= rnum ; ++i) {
+
+            // get next point
+            if (var.cdhr_walk) {
+                rand_coord_prev = rand_coord;
+                rand_coord = uidist(rng);
+                kapa = urdist(rng);
+
+                hit_and_run_coord_update(p, p_prev, P, rand_coord, rand_coord_prev, kapa, lamdas, bpair);
+                newProduct += c(rand_coord) * (bpair.first + kapa * (bpair.second - bpair.first));
+            } else {
+                hit_and_run(p, P, var, p1, p2);
+                newProduct = p.getCoefficients().dot(c);
+            }
+
+
+            // get new minimizing point
+            bool changedMin1 = false;
+            bool changedMin2 = false;
+
+            getNewMinimizingPoints(p, minProduct1, minProduct2, min1, min2, newProduct, changedMin1, changedMin2);
+
+            if (changedMin1){
+                // if the new point is the new min update the boundary point
+
+                boundaryMin2 = boundaryMin1;
+                if (var.cdhr_walk) {
+                    if (c(rand_coord) > 0) {
+                        boundaryMin1 = p_prev;
+                        boundaryMin1.set_coord(rand_coord, boundaryMin1[rand_coord] + bpair.second);
+                    } else {
+                        boundaryMin1 = p_prev;
+                        boundaryMin1.set_coord(rand_coord, boundaryMin1[rand_coord] + bpair.first);
+                    }
+
+                } else {
+                    boundaryMin1 = p1.getCoefficients().dot(c) < p2.getCoefficients().dot(c) ? p1 : p2;
+                }
+            }
+            else if (changedMin2) {
+                if (var.cdhr_walk) {
+                    if (c(rand_coord) > 0) {
+                        boundaryMin2 = p_prev;
+                        boundaryMin2.set_coord(rand_coord, boundaryMin2[rand_coord] + bpair.second);
+                    } else {
+                        boundaryMin2 = p_prev;
+                        boundaryMin2.set_coord(rand_coord, boundaryMin2[rand_coord] + bpair.first);
+                    }
+
+                } else {
+                    boundaryMin2 = p1.getCoefficients().dot(c) < p2.getCoefficients().dot(c) ? p1 : p2;
+                }
+            }
+
+//        assert(P.is_in(min1));
+//        assert(P.is_in(min2));
+//            std::cout << min1.getCoefficients().dot(c) << "\t" << minProduct1 <<"\n";
+        } /*  for (unsigned int i = 1; i <= rnum ; ++i)  */
+
+
+        // find an interior point to start the next phase
+        Point _p =  min1*0.20;
```

Comment:
what are those numbers here? how do you choose them ? should they be exposed in the interface?

Reply:
I choose a new point as a convex combination of four known points - I just need it to be closer to one point, so I chose the coefficients to be 0.4, 0.2, 0.2, 0.2.

Reply:
So the choice of ` 0.4, 0.2, 0.2, 0.2` is arbitrary? Does it come from some experimental evidence? Any idea of how this affects the algorithm?

Reply:
In a later revision, I believe I change them... I just test what works better. My main concern is numerical stability; if the new point is too close to the boundary and I try to check if it's inside the polytope, I get a negative answer, even though I know it is. This affects the ray shooting in hit & run.